components/SearchResult.tsx

```diff
@@ -43,9 +41,32 @@ export default function SearchResult({ searchValue }: { searchValue: string }) {
 
   return (
     <>
-      {data?.results.map((album: Album) => (
-        <Result key={album.id} album={album} />
+      {data?.pages.map((page) => (
+        <>
+          {page.results.map((album: Album) => (
+            <Result key={album.id} album={album} />
+          ))}
+        </>
       ))}
+      {isFetching
+        ? () => {
+            const loadingUiArray = Array(10).fill(null);
+            return (
+              <>
+                {loadingUiArray.map((_, index) => (
+                  <ResultLoading key={index} />
+                ))}
+              </>
+            );
+          }
+        : null}
+      <button
+        className="border-2 border-gray-800 rounded-xl p-3 mb-4"
+        onClick={() => fetchNextPage()}
+        disabled={!hasNextPage || isFetching}
```

Comment:
you conditionally want to hide/show this button in these cases, not disable it.

if there are no more pages and the button is still here but disabled, the user would be confused why it's disabled and might think it's something wrong with there internet, etc.